src/Microsoft.Extensions.Logging.Testing/AssemblyTestLog.cs

```diff
@@ -291,24 +294,18 @@ private class AssemblyLogTimestampOffsetEnricher : ILogEventEnricher
         {
             private DateTimeOffset? _logStart;
 
-            public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+            public AssemblyLogTimestampOffsetEnricher(DateTimeOffset? logStart)
             {
-                if (_logStart.HasValue)
-                {
-                    logEvent.AddPropertyIfAbsent(
-                        propertyFactory.CreateProperty(
-                            "TimestampOffset",
-                            $"{(DateTimeOffset.UtcNow - _logStart.Value).TotalSeconds.ToString("N3")}s"));
-                }
-                else
-                {
-                    _logStart = DateTimeOffset.UtcNow;
-                    logEvent.AddPropertyIfAbsent(
-                        propertyFactory.CreateProperty(
-                            "TimestampOffset",
-                            _logStart.Value.ToString("s")));
-                }
+                _logStart = logStart;
             }
+
+            public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+                => logEvent.AddPropertyIfAbsent(
+                    propertyFactory.CreateProperty(
+                        "TimestampOffset",
+                        _logStart.HasValue
+                            ? $"{(DateTimeOffset.UtcNow - _logStart.Value).TotalSeconds.ToString("N3")}s"
```

Comment:
Related to my earlier comment. You can always use `_logStart.Value.ToUniversalTime()` here.

Reply:
https://msdn.microsoft.com/en-us/library/bb354180(v=vs.110).aspx

> This method converts both DateTimeOffset objects to Coordinated Universal Time (UTC) before calculating the time interval that separates them. This removes any effect that different offsets from UTC may have on the comparison.

I don't think the `ToUniversalTime()` is needed.

Reply:
Interesting. Didn't knew that Subtract did that, but you still need to call `ToUniversalTime()` in other places. 
For example, `logger.LogInformation("Starting test {testName} at {logStart}", testName, logStart.ToString("s"));` (and one more place)...I would say just call `ToUniversalTime()` in all places. This way the code is readable and correct.

Reply:
Ignore my last comment. I see that you are using UtcNow few lines above the code.